src/main/kotlin/net/maizegenetics/phgv2/pathing/DiploidPathFinding.kt

```diff
@@ -0,0 +1,325 @@
+ *      c. Store the path (write hvcf)
+ *
+ * The samples in the keyfile are processed in parallel. Each thread (co-routine) processes a sample. Within sample
+ * processing is not multithreaded.
+ */
+class DiploidPathFinding: CliktCommand(help = "Impute best diploid path using read mappings.") {
+    val pathKeyfile by option(help = "tab-delimited file with first two columns: SampleName, ReadMappingFiles. ReadMappingFiles " +
+            "must be the full path to a read mapping file or a comma separated list of file paths.")
+        .required()
+        .validate { require(File(it).exists()) {"$it is not a valid file"} }
+
+    val hvcfDir by option(help = "The directory containing the hvcf files used to build a HaplotypeGraph for path finding.")
+        .required()
+        .validate { require(File(it).isDirectory) {"$it is not a valid directory."} }
+
+    val referenceGenome by option(help = "path to reference genome (fasta or fastq")
+        .required()
+        .validate { require(File(it).exists()) {"$it is not a valid file"} }
+
+    val outputDir by option(help = "The directory where the output hvcfs will be written. The output file names will be <sampleName>.h.vcf.")
+        .required()
+        .validate { require(File(it).isDirectory) {"$it is not a valid directory."} }
+
+    val probCorrect by option(help = "The probability that a mapped read was mapped correctly")
+        .double()
+        .default(0.99)
+        .validate { require(it in 0.5..1.0) {"prob-correct must be between 0.5 and 1.0"} }
+
+    val probSameGamete by option(help = "The probability of transitioning to the same gamete in the next reference range")
+        .double()
+        .default(0.99)
+
+    val minGametes by option(help = "The minimum number of gametes with a haplotype in a reference range. " +
+            "Reference ranges with fewer gametes will not be imputed.")
+        .int()
+        .default(1)
+        .validate { require(it > -1) {"min-gametes must be a positive integer"} }
+
+    val minReads by option(help = "The minimum number of reads per ReferenceRange. Reference ranges with fewer reads will not be imputed")
+        .int()
+        .default(0)
+        .validate { require(it > -1) {"min-reads must be a positive integer."} }
+
+    val inbreedingCoefficient by option(help = "The estimated coefficient of inbreeding for the samples being evaluated.")
+        .double()
+        .default(0.0)
+        .validate { require(it in 0.0..1.0) {"inbreeding-coefficient must be between 0.0 and 1.0"} }
+
+    val maxReadsPerKb by option(help = "ReferenceRanges with more than max-reads-per-kb will not be imputed.")
+        .int()
+        .default(1000)
+        .validate { require(it > -1) {"max-reads-per-kb must be a positive integer."} }
+
+    val useLikelyAncestors by option(help="Use only the most likely ancestors of each sample for path finding.")
+        .boolean()
+        .default(false)
```

Comment:
Should there be a check or warning in the code if useLikelyAncestors is true, but likelyAncestorFile is blank?  I see you do check if this file is blank in savePath().  And appendParentStats checks if the file exists (can that check for ""?)